### PR TITLE
Minor cleanups and improvements to FileSystemExchange

### PR DESCRIPTION
**Pull request by linzebing**
## Description

Remove redundancy, change misleading config example and improve error surfacing on bootstrap.

## Non-technical explanation

As above.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by arhimondr — APPROVED** [review]

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ else if (ImmutableSet.of("abfs", "abfss").contains(scheme)) {
             configBinder(binder).bindConfig(ExchangeAzureConfig.class);
         }
         else {
-            throw new TrinoException(NOT_SUPPORTED, format("Scheme %s is not supported as exchange spooling storage", scheme));
+            binder.addError(new TrinoException(NOT_SUPPORTED, format("Scheme %s is not supported as exchange spooling storage", scheme)));
```
Comment (losipiuk) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java:71]: what does it change really? Isn't exception caught up the stack anyway and handle same way?
We are throwing in line 52 and this one is no different, right?

Comment (losipiuk) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java:71]: But still:
what does it change really?

Comment (losipiuk) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java:71]: Actually yeah - that way we may collect more than one error.

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ protected void setup(Binder binder)
 
         List<URI> baseDirectories = buildConfigObject(FileSystemExchangeConfig.class).getBaseDirectories();
         if (baseDirectories.stream().map(URI::getScheme).distinct().count() != 1) {
-            throw new TrinoException(CONFIGURATION_INVALID, "Multiple schemes in exchange base directories");
+            binder.addError(new TrinoException(CONFIGURATION_INVALID, "Multiple schemes in exchange base directories"));
```
Comment (losipiuk) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java:52]: If you changed this place you should return already. 
Going forward with some arbitrary schema makes little sense.

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java (hunk 1 above)
Comment (linzebing) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java:72]: I tested and just realized that it doesn't make much of a difference in terms of error surfacing. Added return under line 52. I'm fine both ways.

**Pull request merged**